apm-agent-plugins/apm-jdbc-plugin/src/test/java/co/elastic/apm/agent/jdbc/AbstractJdbcInstrumentationTest.java

```diff
@@ -150,17 +154,34 @@ private void testBatchPreparedStatement() throws SQLException {
         statement.addBatch();
         int[] updates = statement.executeBatch();
         assertThat(updates).hasSize(2);
-        assertSpanRecorded(query, false);
+        assertThat(updates[0]).isEqualTo(1);
+        assertThat(updates[1]).isEqualTo(1);
+        assertSpanRecorded(query, false, 2);
+    }
+
+    private void testMultipleRowsModifiedStatement() throws SQLException {
+        String insert = "INSERT INTO ELASTIC_APM (FOO, BAR) VALUES (3, 'TEST')";
+        Statement statement = connection.createStatement();
+        statement.execute(insert);
+
+        assertThat(reporter.getSpans()).hasSize(1);
+        Db db = reporter.getFirstSpan().getContext().getDb();
+        assertThat(db.getStatement()).isEqualTo(insert);
+        assertThat(db.getAffectedRowsCount())
+            .isEqualTo(statement.getUpdateCount())
```

Comment:
Hmm, it seems all drivers DO support multiple invocations of `getUpdateCount` despite the javadoc. Still not sure if it is safe enough to use it...